src/strings/kmp_substring_search.jl

```diff
@@ -0,0 +1,100 @@
+"""
+    GetIndexWithKMP(string::String, sub_string::String, ignore_case::Bool)::Int
+
+Find if a string contain a given sub string with KMP Search, as an explanation might be too lengthy
+As such, a detailed explanation can be found at https://towardsdatascience.com/pattern-search-with-the-knuth-morris-pratt-kmp-algorithm-8562407dba5b
+
+
+Question
+1. Why KMP instead of naive search?
+- Given n = length of string , m = length of string P
+- Then naive search will have time complexity O(mn) to find all occurrences of pattern P in S
+- However, knuth-morris-pratt (KMP) algorithm will have tim complexity of O(m+n) to find all occurrences of pattern P in S
+- Hence, KMP is more efficient than naive approach
```

Comment:
```suggestion
- Hence, KMP is more efficient than the naive approach
```

Reply:
Thank you for your reply @Panquesito7, I had already made the changes as you requested. Do inform me if there's any changes needed